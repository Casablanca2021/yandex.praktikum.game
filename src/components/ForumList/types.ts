export type Item = {
  id: number,
  name: string,
  answersCount: number,
  createdBy: string,
  date: string,
  category: string,
};

export type Items = Item[];

export type OwnProps = {
  items: Items
}
