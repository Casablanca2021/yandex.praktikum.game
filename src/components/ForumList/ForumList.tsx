import React, { FC } from 'react';
import { Table } from 'semantic-ui-react';
import {t} from 'common/dictionary';
import { OwnProps } from './types';
import { Link } from 'react-router-dom';
import { ROUTES } from 'common/consts';

const ForumList: FC<OwnProps> = (props) => {
  return <Table celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>{t('question')}</Table.HeaderCell>
        <Table.HeaderCell/>
        <Table.HeaderCell/>
      </Table.Row>
    </Table.Header>

    <Table.Body>
      {props.items.map((item, key) => {

        return <Table.Row key = {key}>
          <Table.Cell><Link to={ROUTES.FORUM_VIEW.replace(':id', item.id.toString())}>{item.name}</Link></Table.Cell>
          <Table.Cell>{item.answersCount} answers</Table.Cell>
          <Table.Cell> <strong>{item.category}</strong> <br/> by <strong>{item.createdBy}</strong> <br/> {item.date} </Table.Cell>
        </Table.Row>;
      })}
    </Table.Body>
  </Table>
};
export default ForumList;
