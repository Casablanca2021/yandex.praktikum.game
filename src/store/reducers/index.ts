import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from '../history';
import auth from './auth';
import user from './user';
import ssr from './ssr';

const reducer = combineReducers({
  router: connectRouter(history),
  auth,
  user,
  ssr,
});

export default reducer;
