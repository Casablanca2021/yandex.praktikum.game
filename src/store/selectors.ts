import { AppState } from './types';

export const getAuthSelector = (state: AppState) => state.auth;

export const getUserSelector = (state: AppState) => state.user;

export const getSSRSelector = (state: AppState) => state.ssr;
