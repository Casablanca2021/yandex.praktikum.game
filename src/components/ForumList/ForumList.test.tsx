import ForumList from 'components/ForumList';
import { ConnectedRouter } from 'connected-react-router';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';
import store from 'store';
import history from 'store/history';

describe(`ForumList Component`, () => {
  it(`renders correctly`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <ForumList
              items={[
                {
                  active: true,
                  category: 'cat 1',
                  date: '21.03.2015',
                  name: 'TEST',
                  createdBy: 'me',
                  answersCount: 5,
                  id: 1,
                },
                {
                  active: false,
                  category: 'cat 2',
                  date: '21.01.2015',
                  name: 'TEST 2',
                  createdBy: 'me',
                  answersCount: 5,
                  id: 2,
                },
                {
                  active: true,
                  category: 'cat 3',
                  date: '21.01.2011',
                  name: 'TEST 3',
                  createdBy: 'Arnold',
                  answersCount: 15,
                  id: 2,
                },
              ]}
            />
          </ConnectedRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
