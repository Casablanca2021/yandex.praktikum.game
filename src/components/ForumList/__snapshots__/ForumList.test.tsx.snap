// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumList Component renders correctly 1`] = `
<table
  className="ui celled table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className=""
      >
        Вопрос
      </th>
      <th
        className=""
      />
      <th
        className=""
      />
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
    >
      <td
        className=""
      >
        <a
          href="/forum/1"
          onClick={[Function]}
        >
          TEST
        </a>
      </td>
      <td
        className=""
      >
        5
         answers
      </td>
      <td
        className=""
      >
         
        <strong>
          cat 1
        </strong>
         
        <br />
         by 
        <strong>
          me
        </strong>
         
        <br />
         
        21.03.2015
      </td>
    </tr>
    <tr
      className=""
    >
      <td
        className=""
      >
        <a
          href="/forum/2"
          onClick={[Function]}
        >
          TEST 3
        </a>
      </td>
      <td
        className=""
      >
        15
         answers
      </td>
      <td
        className=""
      >
         
        <strong>
          cat 3
        </strong>
         
        <br />
         by 
        <strong>
          Arnold
        </strong>
         
        <br />
         
        21.01.2011
      </td>
    </tr>
  </tbody>
</table>
`;
